src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -41,6 +50,18 @@ def validate_reservations_summaries(namespace):
         raise CLIError("usage error: Both --start-date and --end-date need to be supplied for daily grain.")
 
 
+def validate_reservations_details_reservationid(namespace):
+    """Validates reservation order id """
+    if not namespace.reservationorderid:
+        raise CLIError("usage error: Reservation Order Id needs to be supplied.")
+    if not namespace.reservationid:
+        raise CLIError("usage error: Reservation Id needs to be supplied.")
+    if not namespace.start_date:
+        raise CLIError("usage error: --start-date needs to be supplied for reservation details.")
+    if not namespace.end_date:
+        raise CLIError("usage error:--end-date needs to be supplied for reservation details.")
```

Comment:
These should not be required.  If they are correctly identified as required parameters, argparse will automatically raise the appropriate error(s)